go/controller/internal/controller/common.go

```diff
@@ -34,14 +35,26 @@ var (
 	waitForResources = ctrl.Result{RequeueAfter: requeueWaitForResources}
 )
 
+// TODO: Remove.
+func notFoundOrReadyErrorMessage(err error) string {
+	return fmt.Sprintf("Referenced object is either not found or not ready, found error: %s", err.Error())
+}
+
 // handleRequeue allows avoiding rate limiting when some errors occur,
 // i.e., when a resource is not created yet, or when it is waiting for an ID.
 //
 // If the result is set, then any potential error will be saved in a condition
 // and ignored in the return to avoid rate limiting.
 //
+// If not found error is detected, then the result is automatically changed to
+// wait for resources.
+//
 // It is important that at least one from a result or an error have to be non-nil.
 func handleRequeue(result *ctrl.Result, err error, setCondition func(condition metav1.Condition)) (ctrl.Result, error) {
+	if err != nil && apierrors.IsNotFound(err) {
```

Comment:
@michaeljguarino This part is responsible for waiting for resources. We discover a `not found` error to enforce requeue

Reply:
gotcha gotcha